src/main/java/seedu/address/logic/commands/FilterCommand.java

```diff
@@ -0,0 +1,40 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_LAB;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_LABSTATUS;
+
+import seedu.address.commons.core.Messages;
+import seedu.address.model.Model;
+import seedu.address.model.person.lab.StudentHasLabPredicate;
+
+public class FilterCommand extends Command {
+
+    public static final String COMMAND_WORD = "filter";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": Finds all students with the specified lab"
+            + " and displays them as a list with index numbers.\n"
+            + "Example: " + COMMAND_WORD + " " + PREFIX_LAB + "1" + " " + PREFIX_LABSTATUS + "s";
+
+    private final StudentHasLabPredicate predicate;
```

Comment:
Are we only filtering by labs? If not maybe just accepting a `Predicate` is fine.

Reply:
There exists a find command to search for names, I was under the assumption that this filter command is exclusively for labs